sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/StorageApiWritesShardedRecords.java

```diff
@@ -640,6 +678,18 @@ public void process(
         }
         appendSplitDistribution.update(numAppends);
 
+        if (updatedSchemaReturned.get() != null) {
+          // Update the table schema and clear the append client.
+          if (appendClientInfo.get().hasSchemaChanged(updatedSchemaReturned.get())) {
+            appendClientInfo.set(
+                AppendClientInfo.of(
+                    updatedSchemaReturned.get(), appendClientInfo.get().getCloseAppendClient()));
+            // TODO: invalidate?
```

Comment:
same as in line 434

Reply:
Same - I believe that put() invalidates the old value

Reply:
added invalidate